### PR TITLE
Override pythons sum() to give the right result for bool tensors

### PR DESCRIPTION
**Pull request by izdeby**
We want pythons sum() to treat boolean tensors the same way as torch.sum()
tested via unit tests

### CODE REVIEW

**Review by gchanan — CHANGES_REQUESTED** [review]
this doesn't look correct.

For example, I don't know why bool would be special but not e.g. uint8.

Also, we already have a definition for `__radd__` -- see torch/csrc/autograd/generated/python_variable_methods.cpp.  What does numpy do here?

File: torch/tensor.py [hunk 1]
```diff
@@ -362,6 +362,12 @@ def __rfloordiv__(self, other):
     __ge__ = _C._TensorBase.ge
     __abs__ = _C._TensorBase.abs
 
+    def __radd__(self, other):
+        if isinstance(self, torch.Tensor):
+            if (self.dtype == torch.bool or self.dtype == torch.uint8):
+                return torch.sum(self) + other
```
Comment (fmassa) [inline torch/tensor.py]: What happens if you do `5 + x`, where `x` is a bool tensor?

Comment (fmassa) [reply torch/tensor.py]: More specifically, if x has more than 1 element, I think this doesn't work as intended

Comment (gchanan) [reply torch/tensor.py]: yep, it's also concerning that this incorrect implementation caused no test errors...

**Comment by ezyang** [comment]
Resigning from review; ping me if you do want me to look

**Pull request closed without merging**